primitives/src/v2/mod.rs

```diff
@@ -609,6 +610,19 @@ pub struct CandidateCommitments<N = BlockNumber> {
 	pub hrmp_watermark: N,
 }
 
+// Custom implementation of `Hash`, since `BoundedVec` does not implement it.
+#[cfg(feature = "std")]
+impl sp_std::hash::Hash for CandidateCommitments {
+	fn hash<H: sp_std::hash::Hasher>(&self, state: &mut H) {
```

Comment:
If we continue needing this, then you could do:

let Self { upward_messages, horizontal_messages, /* listing all fields */ } = self;

This way you get an error if you don't update this, but `CandidateCommitments` gets another field.

Reply:
Great suggestion! I will have to remember to do that in the future.

Reply:
You can also in your other pr just add `#[cfg_attr(feature = "std", derive(Hash))]`

Reply:
@bkchr Nice! For completeness, should I add that for any other bounded types?

Reply:
Yes please.

Reply:
This can now be removed?